test/unit/spec/LoginRouter_spec.js

```diff
@@ -164,13 +165,21 @@ function (Okta, Q, Backbone, SharedUtil, CryptoUtil, CookieUtil, OktaAuth, Util,
       });
     }
 
+    function expectUnexpectedFieldLog(arg1) {
+      expect(Logger.warn).toHaveBeenCalledWith('Field not defined in schema', arg1);
+    }
+
     it('throws a ConfigError if unknown option is passed as a widget param', function () {
+      spyOn(Logger, 'warn');
       var fn = function () { setup({ foo: 'bla' }); };
-      expect(fn).toThrowError(Errors.ConfigError);
+      expect(fn).not.toThrow(Errors.ConfigError);
+      expectUnexpectedFieldLog('foo');
```

Comment:
I'm not sure we should be changing this tests... the test seems to be here to verify that we throw an error when an invalid config param is passed. Changing this now is not throwing that error, but instead it is just logging a warn message, which I'm guessing, is not an exception anymore, but just a warning.

Reply:
@mauriciocastillosilva-okta I made these test changes based on this commit to courage https://github.com/okta/courage/commit/eb4b5baa59e057846fb8784c37bd21be6426dafa. I saw similar test fix in Model_spec.js.

Reply:
right, but is this changing the behavior of the widget? If we pass an invalid config param, what happens now? Do we throw an error so the user knows, or do we just log an error?

Reply:
@rajaguruduraisamy-okta  ^^

Reply:
I believe having extra properties in the courage model will not do any harm. There is no need for this check anymore. We need to check for required and needed params passed alone and any extra unwanted params passed will be discarded right. @rchild-okta ^^ your thoughts.

Reply:
This one is probably fine - we originally made the change in courage to be less strict on API changes, i.e. if the API response returns more properties than are explicitly set in the middle, it wouldn't break the UI. This case is a little different - we have been using this as a hard check to let the user know immediately that they're doing something wrong. But, it will log the error to the console (which is where the original error shows up anyway), so I think that should be sufficient.

Reply:
Oh ok, if it is being output to the user then it is good. I thought for some reason that since we don't throw the exception the error will not be immediately visible to the user.